runtime/Java/src/org/antlr/v4/runtime/CaseChangingCharStream.java

```diff
@@ -0,0 +1,77 @@
+package org.antlr.v4.runtime;
+
+import org.antlr.v4.runtime.misc.Interval;
+
+/**
```

Comment:
Perhaps we could change the comment as follows?

This class supports case-insensitive lexing by wrapping an existing
{@link CharStream} and forcing the lexer to see either upper or
lowercase characters. Grammar literals should then be either upper or
lower case such as 'BEGIN' or 'begin'. The text of the character
stream is unaffected. Example: input BeGiN would match lexer rule
'BEGIN' if constructor parameter upper=true but getText() would return
BeGiN.

Reply:
SGTM.